addons/crm_claim/models/crm_claim.py

```diff
@@ -0,0 +1,130 @@
+    categ_id = fields.Many2one('crm.claim.category', string='Category')
+    priority = fields.Selection([('0', 'Low'), ('1', 'Normal'), ('2', 'High')], default='1')
+    type_action = fields.Selection([('correction', 'Corrective Action'), ('prevention', 'Preventive Action')], string='Action Type')
+    user_id = fields.Many2one('res.users', string='Responsible', track_visibility='always', default=lambda self: self.env.uid)
+    user_fault = fields.Char(string='Trouble Responsible')
+    team_id = fields.Many2one(
+        'crm.team', 'Sales Team', oldname='section_id',
+        index=True, help="Responsible sales team."
+        " Define Responsible user and Email account for"
+        " mail gateway.", default=lambda self: self.env['crm.team']._get_default_team_id())
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env['res.company']._company_default_get('crm.case'))
+    partner_id = fields.Many2one('res.partner', string='Partner')
+    email_cc = fields.Text(string='Watchers Emails', size=252, help="These email addresses will be added to the CC field of all inbound and outbound emails for this record before being sent. Separate multiple email addresses with a comma")
+    email_from = fields.Char(string='Email', size=128, help="Destination email for email gateway.")
+    partner_phone = fields.Char(string='Phone')
+    stage_id = fields.Many2one('crm.claim.stage', 'Stage', track_visibility='onchange', domain="['|', ('team_ids', '=', team_id), ('case_default', '=', True)]", default=lambda self: self.stage_find([], [('sequence', '=', '1')]))
+    cause = fields.Text('Root Cause')
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        """This function returns value of partner address based on partner
+           :param email: ignored
+        """
+        self.email_from = self.partner_id.email
+        self.partner_phone = self.partner_id.phone
+
+    def stage_find(self, cases, domain=[], order='sequence'):
+        """ Override of the base.stage method
+            Parameter of the stage search taken from the lead:
+            - team_id: if set, stages must belong to this team or
+              be a default case
+        """
+        if isinstance(cases, (int, long)):
+            cases = self.browse(cases)
+        search_domain = ([('team_ids', '=', self.team_id.id)]) + ([('team_ids', '=', claim.team_id.id) for claim in cases if claim.team_id])
+        search_domain = ['|'] * len(search_domain) + search_domain + [('case_default', '=', True)] + domain
+        stages = self.env['crm.claim.stage'].search(search_domain, order=order, limit=1)
+        if stages:
+            return stages.id
+        return False
+
+    @api.model
+    def create(self, vals):
+        if vals.get('team_id') and not self.env.context.get('default_team_id'):
+            return super(CrmClaim, self.with_context(default_team_id=vals.get('team_id'))).create(vals)
```

Comment:
wrong. you can assign self.env.context dictionary to one normal dictionary variable  , update that variable value and at last, use with_context with that variable.